core/core_pch.h

```diff
@@ -0,0 +1,3 @@
+#include "exception.h"
```

Comment:
Should decide whether inclusion of MRtrix header files here is to be based on pragmatism (ie. what provides the greatest speedup vs. is least likely to be modified) vs. some sort of principle (ie. what are "the most fundamental" core header files). There might be an argument to be made for `core/*.h` (ie. root of `core/` but not its sub-directories). But I'm not currently able to form a robust opinion on this.

Reply:
I think a nice balance between stability and speed gain should prioritised. Additionally, we should probably include header files that are likely to affect most of the codebase.  For example, this was the reasoning to add `#include "exception.h", since it showed up in `ClangBuilderAnalyzer`'s report and I think there are little chances for a developer to modify this header. Furthermore, this header is indirectly included in the majority of our codebase and thus even without precompiled headers, its modification will trigger a recompilation of a substantial chunk of the project (on this point, it's worth noting that you can always turn off PCHs by setting `cmake -DMRTRIX_USE_PCH=OFF`.
Overall, I do think that we should be very conservative with the inclusion of user headers in a PCH.